src/config.ts

```diff
@@ -0,0 +1,122 @@
+const getInput = (
+	name: string,
+	options?: core.InputOptions,
+): string | undefined => {
+	const got = core.getInput(name, options);
+	return got === '' ? undefined : got;
+};
+
+const readConfigFile = (path: string): object => {
+	const data = read(path);
+	return yaml.load(data) as object;
+};
+
+const getProjectName = ({ stacks }: RiffraffYaml): string => {
+	const appInput = getInput('app');
+	const projectNameInput = getInput('projectName');
+
+	if (!appInput && !projectNameInput) {
+		throw new Error('Must specify either app or projectName.');
+	}
+
+	if (projectNameInput) {
+		return projectNameInput;
+	}
+
+	const numberOfStacks = stacks.length;
+
+	if (numberOfStacks === 1 && appInput) {
+		const stack = stacks[0] as string;
+		return `${stack}::${appInput}`;
+	} else {
+		throw new Error();
+	}
+};
+
+const getRiffRaffYaml = (): RiffraffYaml => {
+	const configInput = getInput('config');
+	const configPathInput = getInput('configPath');
+
+	if (!configInput && !configPathInput) {
+		throw new Error('Must specify either config or configPath.');
+	}
+
+	const configObjFromInput = (
+		configInput
+			? yaml.load(configInput)
+			: readConfigFile(configPathInput as string)
+	) as RiffraffYaml;
+
+	return {
+		/*
+    A valid `riff-raff.yaml` does not need to have `stacks` at the root level, it can be defined within each individual deployment.
+    This action uses the top level `stacks` to create a default project name.
+    This is a little hack to enable this behaviour, else we'd have to start validating `stacks` within each deployment.
+    We create a default project name if and only if `stacks` has a single value.
+     */
+		...{ stacks: [] },
+
+		...configObjFromInput,
+	};
+};
+
+const envOrUndefined = (variableName: string): string | undefined => {
+	const maybeEnvVar = process.env[variableName];
+	return maybeEnvVar && maybeEnvVar.trim() !== ''
+		? maybeEnvVar.trim()
+		: undefined;
+};
+
+const branchName = (): string | undefined => {
+	// `GITHUB_HEAD_REF` is only set for pull request events and represents the branch name (e.g. `feature-branch-1`).
+	// `GITHUB_REF` is the branch or tag ref that triggered the workflow (e.g. `refs/heads/feature-branch-1` or `refs/pull/259/merge`).
+	// Either can be the empty string ¯\_(ツ)_/¯
+	// See https://docs.github.com/en/actions/learn-github-actions/environment-variables
+	const branchName =
+		envOrUndefined('GITHUB_HEAD_REF') ?? envOrUndefined('GITHUB_REF');
+	return branchName ? branchName.replace('refs/heads/', '') : undefined;
+};
+
+const vcsURL = (): string | undefined => {
+	const repoFromEnv = envOrUndefined('GITHUB_REPOSITORY');
+	return repoFromEnv ? `https://github.com/${repoFromEnv}` : undefined;
+};
+
+export interface Configuration {
+	projectName: string;
+	riffRaffYaml: RiffraffYaml;
+	dryRun: boolean;
+	buildNumber: string;
+	branchName: string;
+	vcsURL: string;
+	revision: string;
+	stagingDirInput?: string;
+}
```

Comment:
This represents all configuration used in this app. Values are read from the action input, or environment variables.